charts/consul/values.yaml

```diff
@@ -2503,6 +2514,10 @@ meshGateway:
   # Requirements: consul 1.6.0+ if using `global.acls.manageSystemACLs``.
   enabled: false
 
+  # Override global log verbosity level for mesh-gateway-deployment pods. One of "debug", "info", "warn", or "error".
+  # @type: string
+  logLevel: ""
+
```

Comment:
For any `consul-dataplane` deployment, there are a larger set of options for Envoy's logLevels. Particularly `trace` is important for seeing traffic pass through Envoy. We need to add this to the doc, but we also need to find a way to make sure that we can set `trace` without breaking the control-plane image. I haven't looked into what would happen in this case.

This comment applies to all sections where I noted in the template file there is a dataplane instance (e.g. gateways)

Reply:
We could consider:
Option 1:  separate fields with different log levels, where for consul-k8s components you can set debug/info/warn/error and for dataplane you can set debug/info/warn/error/trace
Option 2: allow consul-k8s components to accept "trace" log level, but internally just use debug for it instead.

Reply:
I like option 2 in this case. Option 1 seems to add a lot of fields to the configuration. @david-yu you have any thoughts here?

Reply:
Option 2 sounds good to me too.

Reply:
I dont think we need to handle anything special in case of `consul-k8s` components wrt log levels. Following are the consul-k8s manifests container the dataplane container

1. ingress-gateways-deployment
2. terminating-gateways-deployment
3. mesh-gateways-deployment
4. telemetry-collector-deployment

All the other files are k8s jobs which don't have a dependency with dataplane. It looks like connect-init already supports trace as a logLevel and I don't see differences between the two flags (in [connect-init](https://github.com/hashicorp/consul-k8s/blob/6624d34b9f1a4d0f12cef156c5f35152ee3e08b5/control-plane/subcommand/connect-init/command.go#L87) and [consul-dataplane](https://github.com/hashicorp/consul-dataplane/blob/2e4715d1939633ed16e4dc06784607f5d926b180/cmd/consul-dataplane/main.go#L41)). I will change the comments in the `values.yaml`

Reply:
Ah yep, consul-k8s does seem to handle trace as a log level through the hclog logger! So it should be fine to pass trace level in.